### PR TITLE
allow region without name in code sample

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/CodeSnippetExtractor/CodeSnippetRegexExtractor.cs [hunk 1]
```diff
@@ -26,6 +26,8 @@ public abstract class CodeSnippetRegexExtractor : ICodeSnippetExtractor
             var result = new Dictionary<string, DfmTagNameResolveResult>();
             foreach (var snippetTagGroup in snippetTagGroups)
             {
+                if (string.IsNullOrEmpty(snippetTagGroup.Key)) continue;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/CodeSnippetExtractor/CodeSnippetRegexExtractor.cs]: >snippetTagGroup [](start = 41, length = 15)

filter the empty key out in line 20?

File: src/Microsoft.DocAsCode.Dfm/CodeSnippetExtractor/RecursiveNameCodeSnippetExtractor.cs [hunk 2]
```diff
@@ -52,6 +51,10 @@ protected override List<CodeSnippetTag> ResolveCodeSnippetTags(string[] lines)
                 var endLineMatch = _endLineRegex.Match(lines[i]);
                 if (endLineMatch.Success)
                 {
+                    if (snippetStack.Count == 0)
+                    {
+                        throw new Exception($"line {i} contains an end line that can't match a start line");
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/CodeSnippetExtractor/RecursiveNameCodeSnippetExtractor.cs]: >line [](start = 46, length = 4)

minor: need we add lines[i] into error message for better trouble shooting?

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/CodeSnippetExtractor/RecursiveNameCodeSnippetExtractor.cs]: Sure, I catch this exception and add a LogError in ResolveTagNamesFromPath.

---
In reply to: [110374469](https://github.com/dotnet/docfx/pull/1528#discussion_r110374469) [](ancestors = 110374469)

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Dfm/CodeSnippetExtractor/CodeSnippetRegexExtractor.cs (hunk 1 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Dfm/CodeSnippetExtractor/CodeSnippetRegexExtractor.cs]: want to keep them in excludedLines. move into foreach for better readability

**Pull request merged**